agent/hcp/telemetry/filter_test.go

```diff
@@ -0,0 +1,39 @@
+package telemetry
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFilter(t *testing.T) {
+	for name, tc := range map[string]struct {
```

Comment:
nit but I noticed we're opting for `t.Parallel()` in other tests, maybe worth doing so here too and in the subtest

Reply:
Yep I definitely should add that, good catch!